soe_profile.profile

```diff
@@ -226,3 +226,35 @@ function soe_profile_config_pages_presave(ConfigPagesInterface $entity) {
     \Drupal::state()->set('xmlsitemap_base_url', $url_field[0]['uri']);
   }
 }
+
+/**
+ * Alter the data of a sitemap link before the link is saved.
+ *
+ * @param array $link
+ *   An array with the data of the sitemap link.
+ * @param array $context
+ *   An optional context array containing data related to the link.
+ */
+function soe_profile_xmlsitemap_link_alter(array &$link, array $context) {
+
+  // Get node/[:id] from loc.
+  $node_id = $link['loc'];
+
+  // Get 403 page path
+  $soe_profile_403_page = \Drupal::config('system.site')->get('page.403');
+
+  // Get 404 page path
+  $soe_profile_404_page = \Drupal::config('system.site')->get('page.404');
+
+  // If node id matches 403 or 404 pages, remove it from sitemap.
+  switch ($node_id) {
+    case $soe_profile_403_page:
+      // Status is set to zero to exclude the item in the sitemap.
+      $link['status'] = 0;
+
```

Comment:
Missing a `break;` here.

Reply:
what if they use the same node for a 404 and 403, the break will stop the other from being set

Reply:
Then perhaps this should be a loop or an if statement as the switch statement is currently leaky. I know if the first condition triggers the remaining all do the same thing so it may be better to just collapse the statement then.

Reply:
ok, i will fix